src/Security/BeamNG.cpp

```diff
@@ -180,17 +180,44 @@ void LegitimacyCheck() {
     RegCloseKey(hKey);
 #elif defined(__linux__)
     struct passwd* pw = getpwuid(getuid());
-    std::string homeDir = pw->pw_dir;
+    std::filesystem::path homeDir = pw->pw_dir;
+
     // Right now only steam is supported
-    std::ifstream libraryFolders(homeDir + "/.steam/root/steamapps/libraryfolders.vdf");
-    auto root = tyti::vdf::read(libraryFolders);
+    std::vector<std::filesystem::path> steamappsCommonPaths = {
+        ".steam/root/steamapps", // default
+        ".var/app/com.valvesoftware.Steam/.steam/root/steamapps", // flatpak
+        "snap/steam/common/.local/share/Steam/steamapps" // snap
+    };
+
+    std::filesystem::path libraryFoldersPath;
+    bool libraryFoldersFound = false;
+
+    for (const auto& path : steamappsCommonPaths) {
+        std::filesystem::path fullPath = homeDir / path / "libraryfolders.vdf";
+        if (std::filesystem::exists(fullPath)) {
+            libraryFoldersPath = fullPath;
+            libraryFoldersFound = true;
+            break;
+        }
+    }
+
+    if (!libraryFoldersFound) {
+        error("Did not find a valid path to a libraryfolders.vdf file.");
+        return;
+    }

```

Comment:
Could this be more something along the lines of "Did not find a supported steam installation"?

Reply:
Sure, but are you sure? In the rare occurrence that the end user has a supported installation but not the .vdf file for some reason, they would be wrongly informed that their Steam installation was not found, adding an extra step in the troubleshooting process for them.
Again, it would be a very rare occurrence, but who knows. Should I account for both scenarios instead? Up to you.

Reply:
Let's do both then, something along the lines of "unsupported steam installation or libraryfolders.vdf is missing"